kaupunginnaiset/alman-akka
# Feat/add navbar to app

## Purpose

This PR adds an initial version of the navbar. Also, I added couple of little things:
- Skip to main content-link, which is visible when focused, otherwise hidden.
- A bit of custom focus styling
- Rainbow flag as a favicon (I needed something to find the correct tab because I have too many tabs open 😅 )

I'm also documenting some things here: 

### Text next to the icons

Having only icons representing something increases cognitive load. Not everyone understands what those icons mean (except maybe, just maybe, for the search icon). Understanding the meaning of icons is based on previous experiences, so some might recognise the icon, and some might not. So, if user doesn't recognise the icon, it increases cognitive load. That's why it's important to have text labels accompanying the icons.

More information from [Nielsen Norman Group](https://www.nngroup.com/articles/icon-usability/).

### Keeping the navigation menu items visible in mobile view as well

From accessibility perspective, it takes work to make mobile navigation (so, the hamburger version) accessible for different types of users. As we don't have many links in the design, I'd like to keep them all visible all the time - at least for now. I think switching the orientation of the link list from row to column in smaller viewports does the trick, at least for all four items that are displayed in the initial designs. 

A note: I added only the ones that are mandatory for the first version. 

### Underlining of the Alman Akka-link

I wanted to make the Alman Akka-link a bit different from the other links in the navbar, and wanted to try out the gradients. Let me know what you think! I used the colors from the progressive pride flag for this. There's a video from the hover/focus effects. This is an initial test, and can definitely be changed to something else 😄 

## Screenshots

Desktop-view:
![Screenshot 2022-03-19 at 10 25 30](https://user-images.githubusercontent.com/28345294/159113718-0f244b58-6e38-4f47-9b7f-01b52f00734f.png)

Mobile-view:
![Screenshot 2022-03-19 at 10 25 41](https://user-images.githubusercontent.com/28345294/159113716-3be93d44-5804-4686-b07c-a9eb8568db64.png)

Video of the skip link, which is visible when focused:


https://user-images.githubusercontent.com/28345294/159114486-b61cdbd1-2bea-4ceb-bded-effd79e9b894.mov



Video of the hover effects:

https://user-images.githubusercontent.com/28345294/159113722-9c299f11-0628-4747-854d-d6118194a252.mov

## Related Issues

This PR resolves #7 

## How to test

There's really not much to test functionally, but you could check the following:

1. Open the page with different viewport widths (for example, using the developer tools' device toolbar)
2. Check, that everything looks ok in different widths.
